Templates/DefaultProject/Template/package.bat

```diff
@@ -0,0 +1,16 @@
+REM --------------------------------------------------------------------------------------------------
+REM 
+REM Copyright (c) Contributors to the Open 3D Engine Project.
+REM For complete copyright and license terms please see the LICENSE at the root of this distribution.
+REM 
+REM SPDX-License-Identifier: Apache-2.0 OR MIT
+REM 
+REM --------------------------------------------------------------------------------------------------
+
+set O3DE_PATH=${EnginePath}
+set O3DE_PROJECT_PATH=${ProjectPath}
+set O3DE_PROJECT_SEEDLIST=%O3DE_PROJECT_PATH%\AssetBundling\SeedLists\Example.seed
+set OUTPUT_PATH=%O3DE_PROJECT_PATH%\ProjectPackages
+
+%O3DE_PATH%\scripts\o3de.bat export-project -es ExportScripts\export_source_built_project.py --project-path %O3DE_PROJECT_PATH% --log-level INFO -assets --build-tools --config release --archive-output zip --seedlist %O3DE_PROJECT_SEEDLIST% -out %OUTPUT_PATH%
```

Comment:
This doesn't work out well at all.
This script would only be usable on the machine of the user who created the project.
Other users would have a different project and engine path and using the hardcoded engine and project path wouldn't be useful at all.

Furthermore this doesn't work if the user who created the project, then later on registers the project with a different engine.

Reply:
@lumberyard-employee-dm I agree that it won't work for many cases but the first and primary use cases is packaging a project that the user created locally. This is an easy first case to solve without diving too deeper into the details of having it run for all cases. 

I can rename it to `sample_package.bat` script if that helps to make it clear.

Reply:
Well, that if instead of a batch script, this was a python script that could determine the project root via using the the same loginc in [utils.get_project_path_from_file](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/utils.py#L511-L525) to determine the project root and then using [manifest.get_project_engine_path](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/manifest.py#L525-L555) to retrieve the engine root.

The current use case here is quite limited to an enthusiast user that wants to package o3de as part of a potential game jam project.

Reply:
The issue is that getting access to that python API requires knowing where the engine is in the first place. So this small example scripts is the simplest form we can do that will be useful in many cases.